pkg/plugin/standalone/v1/dispenser.go

```diff
@@ -0,0 +1,191 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package standalonev1
+
+import (
+	"context"
+	"sync"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/plugin"
+	goplugin "github.com/hashicorp/go-plugin"
+	"github.com/rs/zerolog"
+)
+
+type Dispenser struct {
+	logger zerolog.Logger
+	path   string
+	opts   []ClientOption
+
+	dispensed int
+	client    *goplugin.Client
+	m         sync.Mutex
+}
+
+func NewDispenser(logger zerolog.Logger, path string, opts ...ClientOption) (*Dispenser, error) {
+	d := Dispenser{
+		logger: logger,
+		path:   path,
+		opts:   opts,
+	}
+
+	err := d.initClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &d, nil
+}
+
+func (d *Dispenser) initClient() error {
+	c, err := NewClient(d.logger, d.path, d.opts...)
+	if err != nil {
+		return cerrors.Errorf("could not create go-plugin client: %w", err)
+	}
+
+	d.client = c
+	return nil
+}
+
+func (d *Dispenser) increaseDispensed() error {
+	d.m.Lock()
+	defer d.m.Unlock()
+
+	d.dispensed++
```

Comment:
In which cases is it expected that this will be > 1, i.e. that we will use the same plugin client (and process) for let's say, `specify` and `source`?

Reply:
This is probably overengineered, I don't expect this to ever be over 1 tbh. And it shouldn't, because otherwise the plugin developer needs to care about it in case they are using some shared state.

We can simplify this!